python/ray/serve/_private/replica.py

```diff
@@ -418,26 +483,42 @@ def callable_method_filter(attr):
             return self.callable
         return getattr(self.callable, method_name)
 
-    async def ensure_serializable_response(self, response: Any) -> Any:
-        if isinstance(response, starlette.responses.StreamingResponse):
+    async def send_user_result_over_asgi(self, result: Any, asgi_sender: Send):
+        """Handle the result from user code and send it over the ASGI interface.
 
-            async def mock_receive():
-                # This is called in a tight loop in response() just to check
-                # for an http disconnect.  So rather than return immediately
-                # we should suspend execution to avoid wasting CPU cycles.
-                never_set_event = asyncio.Event()
-                await never_set_event.wait()
+        If the result is already a Response type, it will be sent directly. Else it
+        will be converted to our custom Response type that handles serialization for
+        common Python objects.
+        """
+        if not isinstance(result, (starlette.responses.Response, RawASGIResponse)):
+            await Response(result).send(
+                scope=None, receive=mock_asgi_receive, send=asgi_sender
+            )
+        else:
+            await result(scope=None, receive=mock_asgi_receive, send=asgi_sender)
 
-            sender = ASGIHTTPSender()
-            await response(scope=None, receive=mock_receive, send=sender)
-            return sender.build_asgi_response()
-        return response
+    async def convert_streaming_response(
```

Comment:
[Nit] Rename `convert_streaming_response` -> `buffer_streaming_response`

Reply:
updated to `convert_streaming_response_to_unary`